core/app/models/spree/product_property.rb

```diff
@@ -1,12 +1,20 @@
 # frozen_string_literal: true
 
+require 'discard'
+
 module Spree
   class ProductProperty < Spree::Base
     include Spree::OrderedPropertyValueList
 
     acts_as_list scope: :product
 
-    belongs_to :product, touch: true, class_name: 'Spree::Product', inverse_of: :product_properties
+    acts_as_paranoid
+    include Spree::ParanoiaDeprecations
```

Comment:
If it wasn't under Paranoia before we can avoid calling `acts_as_paranoid` and its deprecations, or is it adding some other value that I can't see at the moment?

Reply:
done https://github.com/solidusio/solidus/pull/3226/files#diff-df2aa0068c69d3f1a851f4d2e7ed3e20R11